### PR TITLE
Flight record

### PR DESCRIPTION
**Pull request by andrespipe**

### CODE REVIEW

File: src/app/components/alert/components/alert-closable/alert-closable.component.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
     class="alert"
     [type]="alert.type"
     (close)="close(alert)">
-    <span class="font-weight-bold" *ngIf="alert.title">{{alert.title}}:</span> {{ alert.message }}
+    <span class="font-weight-bold" *ngIf="alert.title">
```
Comment (juancapm09) [inline src/app/components/alert/components/alert-closable/alert-closable.component.html]: I'd do
`<span>`
     `content`
`</span>`

File: src/app/views/flight-record/flight-record.component.ts [hunk 2]
```diff
@@ -7,9 +19,162 @@ import { Component, OnInit } from '@angular/core';
...
+        if (flightRecord) {
+          this.formSections = mapDBModelToUIModel(flightRecord, this.formSections);
+        } else {
+          this.loadParentInformation();
+        }
+      });
+  }
+
+  loadParentInformation() {
+    this.flightOrderService
```
Comment (juancapm09) [inline src/app/views/flight-record/flight-record.component.ts]: I read on angular's documentation that the best approach for this is to implement a resolver. Check it out.

File: src/app/views/flight-record/flight-record.component.ts [hunk 3]
```diff
@@ -7,9 +19,162 @@ import { Component, OnInit } from '@angular/core';
...
+
+  loadRouteInfo(): boolean {
+    this.currentId = this.route.snapshot.paramMap.get('id');
+    this.flightOrderId = this.route.snapshot.paramMap.get('flightOrderId');
+    this.requirementId = this.route.snapshot.paramMap.get('reqId');
+    return !!this.flightOrderId && !!this.requirementId;
+  }
+
+  loadRecord() {
+    if ( this.currentId ) {
```
Comment (juancapm09) [inline src/app/views/flight-record/flight-record.component.ts]: No spaces inside an if

File: angular.json [hunk 4]
```diff
@@ -57,6 +57,38 @@
...
+            "dev" : {
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.dev.ts"
+                }
+              ],
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
```
[the remark this answers is not in the document]
Comment (andrespipe) [reply angular.json]: Sorry, that's copy paste border effects

File: src/app/views/flight-record/flight-record.component.ts (hunk 2 above)
Comment (andrespipe) [reply src/app/views/flight-record/flight-record.component.ts]: I don't get it....

File: src/app/components/form-controls/components/simpletext/simpletext.component.ts [hunk 5]
```diff
@@ -31,7 +31,10 @@ export class SimpletextComponent implements OnInit {
   }
 
   validate(): void {
-    this.formInput.isValid = this.isValidated && !!this.model || !this.isValidated;
+    if (typeof this.model === 'number') {
```
[the remark this answers is not in the document]
Comment (andrespipe) [reply src/app/components/form-controls/components/simpletext/simpletext.component.ts]: No..... Sorry but I had a problem here. 0 could be a valid id or value.

File: src/app/views/flight-record/flight-record.component.ts (hunk 2 above)
Comment (andrespipe) [reply src/app/views/flight-record/flight-record.component.ts]: Let's improve in another branch

**Pull request merged**